apps/prairielearn/src/lib/copy-question.ts

```diff
@@ -19,7 +19,7 @@ const sql = sqldb.loadSqlEquiv(import.meta.url);
 export async function setQuestionCopyTargets(res: Response) {
   // Avoid querying for editable courses if we won't be able to copy this
   // question anyways.
-  if (!res.locals.course.template_course) {
+  if (!(res.locals.course.template_course || res.locals.question.shared_publicly_with_source)) {
```

Comment:
Total nit: IMO the version of this where `!` is distributed is easier to parse, what do you think?

```suggestion
  if (!res.locals.course.template_course && !res.locals.question.shared_publicly_with_source) {
```

Ditto for the similar conditional on the `instructorCopyPublicQuestion` page. Feel free to ignore if you disagree.